python/instrumentation/openinference-instrumentation-openai/src/openinference/instrumentation/openai/_request.py

```diff
@@ -302,8 +302,12 @@ def __call__(
             return wrapped(*args, **kwargs)
         try:
             cast_to, request_parameters = _parse_request_args(args)
-            # E.g. cast_to = openai.types.chat.ChatCompletion => span_name = "ChatCompletion"
-            span_name: str = cast_to.__name__.split(".")[-1]
+            # Use consistent span names: "CreateEmbeddings" for embeddings, class name for others
+            if cast_to is self._openai.types.CreateEmbeddingResponse:
+                span_name = "CreateEmbeddings"
```

Comment:
just for my understanding here - what does the consistent name buy us rather than maybe naming based on function name.

Reply:
backends typically index the span name field for query and sometimes people do aggregates/auto-complete on the span name. For example, zipkin does auto-complete on span name, but not arbitrary attributes. Some tools dereive metrics from spans and need to aggregate on something, typically that's a span name. So, if something that is in a spec leaves the span name out, it ends up unable to be usable for things like this.

Reply:
I see. I mainly ask because I tend to face a fair amount of pressure to cram things like agent operations in names because there's a need for operators to groc the control flow.

With embedding's generation, though I think this makes a lot of sense

Reply:
Agreed. I don't think agent ops are commodity yet maybe not for a long while. Wont go trying to normalize those ;)